### PR TITLE
loading icon added

### PR DESCRIPTION
**Pull request by samscupoftea**
1. I added the loading indicator and improved the overall styling via CSS.
2. The loading indicator is triggered in .finally() call at the end of fetch. 
3. resultElement.innerHTML = ''; was added to the start of the button event listener. This resets the recipe to disappear after each click to only show the loading indicator while loading. 
4. I also added a slight black gradient to the background that we can dim further for better accessibility and view-ability. 
Screenshot 2024-04-16 at 18 05 24
Screenshot 2024-04-16 at 18 05 18
Screenshot 2024-04-16 at 18 04 40

### CODE REVIEW

File: public/index.css [hunk 1]
```diff
@@ -161,20 +163,26 @@ button {
   margin-bottom: 1em;
 }
 #loading-indicator {
-  border: 5px solid rgb(73, 180, 87); 
-  border-top: 5px solid rgb(0, 0, 0); 
+  border: 5px solid transparent;
```
Comment (samscupoftea) [inline public/index.css:166]: I made improvements to the styling of the the loading indicator and fixed the text spinning issue below.

File: public/index.js [hunk 2]
```diff
@@ -33,6 +34,8 @@ buttons.forEach((button) => {
   button.addEventListener("click", () => {
     console.log(lactoseIntolerant.checked);
     dishOriginCountry = button.value;
+    loadingContainer.style.display = "block";
```
Comment (samscupoftea) [inline public/index.js:37]: @josueJURE this sets the loadingIndcator style to block, making it appear once button is clicked.

File: public/index.js [hunk 3]
```diff
@@ -33,6 +34,8 @@ buttons.forEach((button) => {
...
+    resultElement.innerHTML = '';
```
Comment (samscupoftea) [inline public/index.js:38]: This resets the previously viewed recipe to disappear each time to only show the loading indicator. @josueJURE

File: public/index.js [hunk 4]
```diff
@@ -78,7 +82,11 @@ buttons.forEach((button) => {
           console.error('Error: Element with class "gpt-response" not found');
         }
       })
-      .catch((error) => console.error("Error:", error));
+      .catch((error) => console.error("Error:", error))
+      .finally(() => {
```
Comment (samscupoftea) [inline public/index.js:86]: Added a .finally which - once the recipe as loaded - triggers the loader to disappear. 
@josueJURE

File: public/index.css [hunk 5]
```diff
@@ -25,6 +25,7 @@
     border: black 1px solid;
     border-radius: 5%;
     background-image: url('./image/dish-image.jpeg'); 
+    background-image: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url('./image/dish-image.jpeg');
```
Comment (samscupoftea) [inline public/index.css:28]: Added a slight gradient that we can increase of decrease to the background image for accessibility and aesthetic purposes. @josueJURE

File: public/index.css (hunk 1 above)
Comment (josueJURE) [reply public/index.css:166]: great

**Pull request merged**